build.sbt

```diff
@@ -57,7 +57,8 @@ lazy val root = (project in file("."))
       "-Wunused:patvars", // Warn if a variable bound in a pattern is unused.
       "-Wvalue-discard", // Warn when non-Unit expression results are unused.
       "-Ycache-plugin-class-loader:last-modified", // Enables caching of classloaders for compiler plugins
-      "-Ycache-macro-class-loader:last-modified" // and macro definitions. This can lead to performance improvements.
+      "-Ycache-macro-class-loader:last-modified", // and macro definitions. This can lead to performance improvements.
+      "-Xfatal-warnings" // Fail the compilation if there are any warnings.
```

Comment:
Usually not a fan of this because when you're hacking about on a problem you'll have to constantly clean up